lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,31 @@
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+use Time::Seconds;
+
+my $DEFAULT_STATE_FILE = "/var/lib/openqa/share/gitlab_maintenance.lock";
```

Comment:
This should not be under `…/share/…` and the `/var/lib` part should not be hard-coded. I suggest you follow the same pattern as in `lib/OpenQA/Schema/ResultSet/Assets.pm` (except for using the `cache` subdirectory).